### DISCUSSION TITLE
Is there any plan to implement EF-VI type of capability for the driver?

### DISCUSSION THREAD

**Discussion by npone** [corundum/corundum#41] [Ideas]
Jus wondering since that would open a lot of possibilities for use cases.

**Comment by alexforencich** [comment]
Never heard of EF-VI before.  But, we're certainly open to adding support for new features, including complex ones, if it makes sense to do so and there is enough interest.  Can you point to any reference information on EF-VI, including existing open implementations?  Does it require any particular hardware capabilities?

**Reply by npone** [reply]
Basically it's Solarflare's "virtual interface" to let user create their own socket and send/receive raw packets directly to/from the network device. No hardware change is need as far as I can see. It's mostly driver work.

Here is where I learned about it:
https://github.com/majek/openonload/blob/master/README.ef_vi
This is  https://usermanual.wiki/Pdf/SF114063CD5efviUserGuide.1954759386/help

**Reply by alexforencich** [reply]
Seems pretty specific to solarflare.  Do any other devices support that interface?  What are the advantages of EF-VI vs. other userspace networking techniques?

**Reply by npone** [reply]
The only other userspace networking technique I have tried is using raw socket, and it's not very easy to use. I'm not familiar with other user space networking techniques. A quick search returned DPDK, but does it work with mqnic? Obviously if there are readily available solutions that work my suggestion basically would be invalid. I'd love hear your insights.
